File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    arr = [0] * len(prices)
    #주어진 길이만큼의 배열을 0으로 초기화

    before = 0

    for i in range(len(prices)):
        cnt = 0
        for k in range(i+1,len(prices)):
            if prices[i]>prices[k]:
                cnt +=1
                break
            cnt+=1
        arr[i] = cnt

    return arr

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];

        for (int i = 0; i < prices.length-1; i++) {
            int cnt = 0;
            for(int k=i+1; k<prices.length; k++){
                if(prices[i] <= prices[k]){
                    cnt++;
                }else{
                    cnt++;
                    break;
                }
            }
            answer[i] = cnt;
        }

        return answer;
    }
}